implementation/jwt-jaxrs/src/main/java/io/smallrye/jwt/auth/jaxrs/SmallRyeJWTAuthJaxRsFeature.java

```diff
@@ -48,11 +46,7 @@ public boolean configure(FeatureContext context) {
         if (enabled) {
             context.register(JWTAuthorizationFilterRegistrar.class);
 
-            if (!SmallRyeJWTAuthCDIExtension.isHttpAuthMechanismEnabled()) {
-                context.register(JWTAuthenticationFilter.class);
-
-                JAXRSLogging.log.eeSecurityNotInUseButRegistered(JWTAuthenticationFilter.class.getSimpleName());
-            }
+            context.register(JWTAuthenticationFilter.class);
```

Comment:
@sberyozkin - I'm trying to get my head around the impact of registering the filter unconditionally. Suppose the HttpAuthMechanism is registered via the CDI extension. I assume (but I am not certain) that the filter would then execute second, and the principal would be an instance of `JsonWebToken`, bypassing the code in `filter`. What are your thoughts?

Reply:
Hi @MikeEdgar I think it would be simplest to copy that code from the feature checking for `HttpAuthMechanism` just to preserve the original logic. We can fine tune it further as required... Otherwise, indeed, I'm not 100% sure of the side-effects introduced by this update - it does not look like this feature is used right now, but it is better to keep the check in place

Reply:
@MikeEdgar I've tried to do this copy but found it was not easy to do. I suppose what I said in the closed PR probably makes sense. The user should be now responsible for making sure no ambiguities happen.
This JAX-RS feature is not annotated as `Provider` and it is not loaded in the CDI extension. So if some users would like to use this feature - they'd have to enable it in their `Application` or via `DynamicFeature`.
And if the users choose to use the CDI extension then all is good - it will load the HTTP mechanism first if EESec is enabled and if not - it will register the JAX-RS filter directly.
So I think all is good :-) - do you agree ?

Reply:
@sberyozkin - I agree. Also, for historical perspective, much of this code was originally intended to allow using SmallRye JWT in Wildfly prior to it's official inclusion.

Reply:
@MikeEdgar OK, thanks. Ok then I'll go ahead with the merge now that Darran also checked it for us to do RC2 today